python/ray/autoscaler/commands.py

```diff
@@ -131,19 +131,21 @@ def kill_node(config_file, yes, override_cluster_name):
         nodes = provider.non_terminated_nodes({TAG_RAY_NODE_TYPE: "worker"})
         node = random.choice(nodes)
         logger.info("kill_node: Terminating worker {}".format(node))
-
-        updater = NodeUpdaterThread(
-            node_id=node,
-            provider_config=config["provider"],
-            provider=provider,
-            auth_config=config["auth"],
-            cluster_name=config["cluster_name"],
-            file_mounts=config["file_mounts"],
-            initialization_commands=[],
-            setup_commands=[],
-            runtime_hash="")
-
-        _exec(updater, "ray stop", False, False)
+        if hard:
+            provider.terminate_node(node)
```

Comment:
Why is this way of killing the node not used by default? Is it because it can possibly block indefinitely?

Reply:
Because killing the raylet is faster for some previous fault tolerance tests - should this be a separate PR?

Reply:
also cc @stephanie-wang